packages/apex-node/test/coverageReporters/coverageReporter.test.ts

```diff
@@ -0,0 +1,261 @@
+          49,
+          50,
+          51,
+          52,
+          54,
+          57,
+          58,
+          59,
+          60
+        ],
+        uncoveredLines: []
+      }
+    },
+    {
+      attributes: {
+        type: 'ApexCodeCoverageAggregate',
+        url:
+          '/services/data/v54.0/tooling/sobjects/ApexCodeCoverageAggregate/71511000006fb7mAAA'
+      },
+      ApexClassOrTrigger: {
+        attributes: {
+          type: 'Name',
+          url:
+            '/services/data/v54.0/tooling/sobjects/ApexClass/01p1100000TqGm7AAF'
+        },
+        Id: '01p1100000TqGm7AAF',
+        Name: 'PropertyController'
+      },
+      NumLinesCovered: 41,
+      NumLinesUncovered: 3,
+      Coverage: {
+        coveredLines: [
+          2,
+          3,
+          16,
+          25,
+          27,
+          28,
+          29,
+          30,
+          32,
+          33,
+          35,
+          36,
+          38,
+          39,
+          40,
+          41,
+          45,
+          46,
+          47,
+          48,
+          49,
+          50,
+          52,
+          67,
+          68,
+          69,
+          70,
+          71,
+          72,
+          75,
+          76,
+          87,
+          88,
+          92,
+          97,
+          98,
+          101,
+          103,
+          104,
+          107,
+          110
+        ],
+        uncoveredLines: [26, 31, 78]
+      }
+    },
+    {
+      attributes: {
+        type: 'ApexCodeCoverageAggregate',
+        url:
+          '/services/data/v54.0/tooling/sobjects/ApexCodeCoverageAggregate/71511000006fb7nAAA'
+      },
+      ApexClassOrTrigger: {
+        attributes: {
+          type: 'Name',
+          url:
+            '/services/data/v54.0/tooling/sobjects/ApexClass/01p1100000TqGm4AAF'
+        },
+        Id: '01p1100000TqGm4AAF',
+        Name: 'GeocodingService'
+      },
+      NumLinesCovered: 36,
+      NumLinesUncovered: 0,
+      Coverage: {
+        coveredLines: [
+          5,
+          8,
+          10,
+          11,
+          12,
+          13,
+          14,
+          15,
+          16,
+          17,
+          18,
+          19,
+          20,
+          21,
+          22,
+          23,
+          24,
+          25,
+          26,
+          28,
+          29,
+          30,
+          31,
+          32,
+          33,
+          34,
+          35,
+          36,
+          38,
+          39,
+          40,
+          41,
+          42,
+          44,
+          48,
+          50
+        ],
+        uncoveredLines: []
+      }
+    }
+  ]
+};
+
+describe('coverageReports', async () => {
+  let testResultsDir: string;
+
+  before(() => {
+    testResultsDir = path.join(
+      tmpdir(),
+      crypto.randomBytes(10).toString('hex')
+    );
+  });
+  after(async () => {
+    try {
+      await fs.promises.rmdir(testResultsDir, { recursive: true });
+    } catch (err) {}
+  });
+  it('should produce coverage report', async () => {
```

Comment:
Could you add some additional tests for optional attributes? For instance, if options (CoverageReporterOptions) is omitted from the call to the constructor, and also if Coverage is missing on the ApexCodeCoverageRecord.

Reply:
Will do.

Reply:
Those are looking swell - thanks for the quick add!